models/cartsDao.js

```diff
@@ -289,6 +285,57 @@ const queryDeleteAllItemInCart = async (userId) => {
   }
 }
 
+const queryUserOrder = async (userId) => {
+  try {
+    const createOrder = await database.query(
+      `
+      INSERT INTO
+        orders(user_id)
+      VALUES(?)
+       `,
+      [userId]
+    )
+
+    const getOrder = await database.query(
+      `SELECT
+        o.id,
+        sc.user_id,
+        sc.product_id,
+        sc.quantity,
+        p.product_price 
+      FROM shopping_carts AS sc
+      JOIN orders AS o ON sc.user_id = o.user_id
+      JOIN products AS p ON sc.product_id = p.id
+      WHERE
+        o.user_id = ?        
+        `,
+      [userId]
+    )
+
+    for (const order of getOrder) {
+      const createOrderItems = await database.query(
+        `
+        INSERT INTO order_items (orders_id, product_id, order_item_price, order_item_quantity)
+        VALUES (?, ?, ?, ?)
+        `,
+        [order.id, order.product_id, order.product_price, order.quantity]
+      )
+    }
```

Comment:
- 현재 For 문이 돌 때마다 건 바이 건으로 database query를 호출하는 비효율적인 구조로 되어있습니다.
- 미리 내가 원하는 데이터의 배열을 만들어 놓고 한번에 삽입하는 bulk_insert의 개념을 아래와 같이 작성해보세요!
- 또한 가독성 좋게 for 문을 돌림과 동시에 배열에 해당 데이터를 잘 쌓아주는 map 함수를 이용하여 리팩토링해보세요!

```suggestion
const orderItems = getOrder.map((order) => [
  order.id,
  order.product_id,
  order.product_price,
  order.quantity
]);

await database.query(`
  INSERT INTO 
  order_items (
    orders_id, 
    product_id, 
    order_item_price, 
    order_item_quantity)
  VALUES ?
`, [orderItems]);

```